services/node_stake_update.proto

```diff
@@ -0,0 +1,69 @@
+syntax = "proto3";
+
+package proto;
+
+/*-
+ * ‌
+ * Hedera Network Services Protobuf
+ * ​
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+option java_package = "com.hederahashgraph.api.proto.java";
+option java_multiple_files = true;
+
+import "basic_types.proto";
+import "timestamp.proto";
+
+/**
+ * Updates the staking info at the end of staking period to indicate new staking period has started.
+ */
+message NodeStakeUpdateTransactionBody {
+  /**
+   * Time and date of the end of the staking period that is ending
+   */
+  Timestamp end_of_staking_period = 1;
+
+  /**
+   * Number of rounds during the staking period that is ending
+   */
+  uint32 num_staking_rounds = 2;
+
+  /**
+   * Staking info of each node at the beginning of the new staking period
+   */
+  repeated StakingNodeInfo staking_node_info = 3;
+}
+
+/**
+ * Staking info for each node for the staking period that is ending
+ */
+message StakingNodeInfo {
+  /**
+   * Calculated consensus weight for this node throughout the current staking period.
```

Comment:
should this be the consensus weight for the ending staking period? `stake_rewarded_start` description says `at the end of staking period` btw

Reply:
Fixed. Thanks